internal/prefixcollector/collector.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package prefixcollector contains excluded prefix collector, prefix sources and functions working with them
+package prefixcollector
+
+import (
+	"cmd-exclude-prefixes-k8s/internal/utils"
+	"context"
+	"io/ioutil"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/excludedprefixes"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/prefixpool"
+)
+
+// ExcludePrefixSource is source of excluded prefixes
+type ExcludePrefixSource interface {
+	Start(chan<- struct{})
```

Comment:
let's return GetUpdateChannel() chan <- string[] ??

Reply:
If we use channels instead of slice, excluded prefixes updates/removals monitoring will be more complicated, because we will need to distinguish prefixes from different sources in final prefixes list, which we write to file.